lib/biran/erb_config.rb

```diff
@@ -1,34 +1,56 @@
 module Biran
   class ERBConfig
-    attr_reader :output_dir, :source_dir, :name, :extension, :config, :output_name
-    attr_accessor :bindings
+    attr_reader :output_dir, :source_dir, :name, :extension, :config, :template_contents
+    attr_accessor :bindings, :output_name, :template_config_index
+
+    DEFAULT_TEMPLATE_CONFIG_INDEX = 1
 
     def initialize(config, name, extension, source, output_dir, output_name)
-      @name       = name
-      @extension  = extension
-      @config     = config
-      @source_dir = source
-      @output_dir = output_dir
-      @output_name = output_name
+      before_process_erb do
+        @name       = name
+        @extension  = extension
+        @config     = config
+        @source_dir = source
+        @output_dir = output_dir
+        @output_name = output_name
+      end
     end
 
     def save!
       File.open(File.join(output_dir, "#{output_name}#{extension}"), 'w') do |f|
-        f.print process_erb.result(build_erb_env.call)
+        f.print template_contents.result(build_erb_env.call)
       end
     end
 
     private
 
+    def before_process_erb
+      yield
+      raise ArgumentError unless process_erb_ready?
```

Comment:
I think this is cleaner for now. If you are getting this error, most likely, you are working on biran and should know what the issue is or at least where to figure it out. As long as all the objects are set, should be good, as the error checking in process erb will catch if the template file is missing and protect against the empty values.